server/controllers/session-ctrl.js

```diff
@@ -0,0 +1,40 @@
+const Session = require("../models/session-model").model;
+
+const generateSessionId = () => {
+  const arr = Array.from({ length: 8 }, () => Math.floor(Math.random() * 256));
+  return Array.from(arr, val => val.toString(16)).join("");
+};
+
+const getUniqueSessionId = async () => {
+  let id = generateSessionId();
+  while (await Session.exists({ _id: id })) {
```

Comment:
```suggestion
  while (!(await Session.exists({ _id: id }))) {
```

oops I made a mistake

Reply:
wait no it was fine lol